app/services/usps_in_person_proofing/date_validator.rb

```diff
@@ -0,0 +1,36 @@
+module UspsInPersonProofing
+  # Validator that can be attached to a form or other model
+  # to verify that specific supported fields are comparable to other dates.
+  # Since it's a subclass of ComparisonValidator, it share the same options.
+  # == Example
+  #
+  #   validates_with UspsInPersonProofing::DateValidator,
+  #     attributes: [:dob],
+  #     message: "error",
+  #     less_than_or_equal_to: ->(_rec) {
+  #         Time.zone.today - IdentityConfig.store.idv_min_age_years.years
+  #     }
+  #     ....
+  #
+  class DateValidator < ActiveModel::Validations::ComparisonValidator
+    private
+
+    def prepare_value_for_validation(value, _record, _attr_name)
+      begin
+        val_to_date(value)
+      rescue
+        nil
+      end
+    end
```

Comment:
we can use the method-level rescue for this instead of needing an explicit `begin`
```suggestion
    def prepare_value_for_validation(value, _record, _attr_name)
      val_to_date(value)
    rescue
      nil
    end
```

Reply:
Ah, so many ways to do a thing.